File: cpp/MNN/test/test_gelu.py
import sys
# pwd 
import os
current_working_directory = os.getcwd()
print("Current working directory:", current_working_directory)

sys.path.append("../python/")  # Adjust the path as necessary
from MNN import GELUActivation
import numpy as np


model = GELUActivation()

input = np.random.uniform(-1, 1, (1, 5, 768)).astype(np.float32)

output = model.forward(input)
print("Input shape:", input.shape)
print("Output shape:", output.shape)
save_path = "build/test/temp"
input.tofile(os.path.join(save_path, "input.bin"))
output.tofile(os.path.join(save_path, "output.bin"))

File: cpp/MNN/test/test_gelu.cpp
#include "MActivation.hpp"
#include "test_common.hpp"

using namespace mnn;
using namespace mtb;

#define FP_T float

void test_MGELUActivation(){
    START_TEST();
    auto input = load_data<FP_T>(
      "build/test/temp/input.bin", 1 * 5 * 768);
    auto output = load_data<FP_T>(
      "build/test/temp/output.bin", 1 * 5 * 768);
    MGELUActivation<FP_T> at;
    
    Tensor<FP_T> input_tensor({1, 5, 768}, input);
    Tensor<FP_T> result_tensor = at.forward(input_tensor);

    compare_data<FP_T>(result_tensor.data().get(), 
                        output.data(), 1 * 5 * 768, 1e-5f);

    PASSLOG();
}

int main(int argc, char** argv) {
    test_MGELUActivation();
    return 0;
}